File: catkin_ws/src/my_robot_tutorials/scripts/numberPubSubReset.py
#! /usr/bin/env python

import rospy
from std_msgs.msg import Int64
from std_srvs.srv import SetBool

trigger = False
value = 0

def handle_count_reset(req:SetBool._request_class):
    global value
    rospy.loginfo(f"Recieved reset service request")
    value = 0
    return {"success":True, "message":"Reset Complete"}

def cb_recieve_data(data:Int64):
    global value
    if not value:
        rospy.loginfo(f"Seed recieved {data.data}")
        value = data
    else:
        value.data += 1
        rospy.loginfo(f"Counter incrementing: {value.data}")
        

def cb_recieve_count(data:Int64):
    rospy.loginfo(f"Counter: {data.data}")

if __name__ == "__main__":
    rospy.init_node("number_pubsub")

    sub_seed = rospy.Subscriber("/seed", 
                           Int64, 
                           cb_recieve_data)
    
    sub_counter = rospy.Subscriber("/counter",
                                   Int64,
                                   cb_recieve_count)
    
    pub = rospy.Publisher("counter_count", Int64, queue_size=1)
    reset_service = rospy.Service("/reset_counter", SetBool, handle_count_reset)
    rospy.loginfo("Counter reset service started")

    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        pub.publish(value)
        rate.sleep()
    rospy.loginfo("Node is exiting")

File: catkin_ws/src/my_robot_tutorials/src/numberPubSubReset.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>
#include <std_srvs/SetBool.h>


std_msgs::Int64 value;
bool handle_counter_reset(std_srvs::SetBool::Request &request,
                        std_srvs::SetBool::Response &response)
{
    ROS_INFO("Reset counter request");
    value.data = 0;
    response.message = "Reset complete";
    response.success = true;
    return true;
}

void cb_recieve_seed(const std_msgs::Int64 &seed)
{
    if(value.data==0)
    {
        ROS_INFO("Seed recieved: %i", (int)seed.data);
        value.data = seed.data;
    }
    else
    {
        value.data += 1;
        ROS_INFO("Incrementing counter %i", (int)value.data);
    }
}

void cb_recieve_count(const std_msgs::Int64 &count)
{
    ROS_INFO("Counter: %i", (int)count.data);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "number_pubsub");
    ros::NodeHandle nh;
    ros::Subscriber sub_seed = nh.subscribe("/seed",
                                            1,
                                            cb_recieve_seed);

    ros::Subscriber sub_counter = nh.subscribe("/counter",
                                                1,
                                                cb_recieve_count);

    ros::Publisher pub = nh.advertise<std_msgs::Int64>("/counter", 1);
    ros::ServiceServer resetServer = nh.advertiseService("/reset_counter",
                                                    handle_counter_reset);
    ros::spin();

}